packages/kbn-user-profile-components/src/user_profiles_selectable.tsx

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { EuiSelectableOption, EuiSelectableProps } from '@elastic/eui';
+import {
+  EuiButtonEmpty,
+  EuiFieldSearch,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiPanel,
+  EuiSelectable,
+  EuiSpacer,
+  EuiText,
+  EuiTextColor,
+} from '@elastic/eui';
+import type { FunctionComponent } from 'react';
+import React, { useEffect, useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { getUserDisplayName } from './imported_types/user';
+import type { UserProfile, UserProfileAvatarData } from './imported_types/user_profile';
+import { UserAvatar } from './user_avatar';
+
+export type UserProfileWithAvatar = UserProfile<{ avatar?: UserProfileAvatarData }>;
+
+export interface UserProfilesSelectableProps
+  extends Pick<EuiSelectableProps, 'height' | 'singleSelection'> {
```

Comment:
It's fairly common to build a specialised version of a component that essentially just forwards on props to the wrapped component using rest/spread. Avoiding Pick/Omit here isn't really feasible since it would lead to a huge amount of duplication. We wouldn't have to only duplicate the types but also the jsdocs for each type which puts an unrealistic burden on maintainers to keep these in sync. I would suggest to update our tooling to be able to deal with these utility types instead of pushing this overhead onto maintainers.